architecture/src/iosMain/kotlin/navigation/Navigator.kt

```diff
@@ -118,15 +118,22 @@ actual class Navigator<A : NavigationAction<*>>(parentVC: UIViewController, priv
 
     private fun embedNestedViewController(nestedSpec: NavigationSpec.Nested) {
         val parent = parent.get() ?: return
-        if (nestedSpec.type == NavigationSpec.Nested.Type.Replace) {
-            parent.childViewControllers.map { it as UIViewController }.forEach {
-                it.willMoveToParentViewController(null)
-                it.view.removeFromSuperview()
-                it.removeFromParentViewController()
+        when (val type = nestedSpec.type) {
+            is NavigationSpec.Nested.Type.Replace -> {
+                parent.childViewControllers.map { it as UIViewController }.filter { it.view.tag == type.tag }.forEach {
```

Comment:
So if you have (accidentally and very possible situation) some child view controller with the same view tag it will removed

Reply:
The alternative is removing everything. So I think using tags is a fine approach. Android handles things similarly for fragments